url7b90c1010b_DastanIqbal_HelloZMQwithJava_tgz-pJ8-ClientJ8/src/zmq/TcpAddress.java

```diff
@@ -0,0 +1,120 @@
+/*
+    Copyright (c) 2009-2011 250bpm s.r.o.
+    Copyright (c) 2007-2009 iMatix Corporation
+    Copyright (c) 2007-2011 Other contributors as noted in the AUTHORS file
+
+    This file is part of 0MQ.
+
+    0MQ is free software; you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    0MQ is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package zmq;
+
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.UnknownHostException;
+
+public class TcpAddress implements Address.IZAddress {
+
+    public static class TcpAddressMask extends TcpAddress {
+        public boolean match_address(SocketAddress addr_) {
+            return address.equals(addr_); 
+        }
+    }
+
+    protected InetSocketAddress address;
+    
+    public TcpAddress(String addr_) {
+        resolve(addr_, false);
+    }
+    public TcpAddress() {
+    }
+    
+    @Override
+    public String toString() {
+        if (address == null) {
+            return null;
+        }
+        
+        if (address.getAddress() instanceof Inet6Address) {
+            return "tcp://[" + address.getAddress().getHostAddress() + "]:" + address.getPort();
+        } else {
+            return "tcp://" + address.getAddress().getHostAddress() + ":" + address.getPort();
+        }
+        
+    }
+    
+
+// JDK6    @Override
+    public void resolve(String name_, boolean ipv4only_) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 95-111 into a separate method.